pkg/handler/repository_parameters.go

```diff
@@ -12,15 +12,49 @@ import (
 )
 
 type RepositoryParameterHandler struct {
-	RepositoryDao dao.RepositoryConfigDao
+	RepositoryDao       dao.RepositoryConfigDao
+	ExternalResourceDao dao.ExternalResourceDao
 }
 
-func RegisterRepositoryParameterRoutes(engine *echo.Group, repoDao *dao.RepositoryConfigDao) {
-	rph := RepositoryParameterHandler{RepositoryDao: *repoDao}
+func RegisterRepositoryParameterRoutes(
+	engine *echo.Group,
+	repoDao *dao.RepositoryConfigDao,
+	externalDao *dao.ExternalResourceDao,
+) {
+	rph := RepositoryParameterHandler{RepositoryDao: *repoDao, ExternalResourceDao: *externalDao}
 	engine.GET("/repository_parameters/", rph.listParameters)
+	engine.GET("/repository_parameters/external_gpg_key/", rph.fetchGpgKey)
 	engine.POST("/repository_parameters/validate/", rph.validate)
 }
 
+// FetchGpgKeys godoc
+// @Summary      Fetch gpgkey from URL
+// @ID           fetchGpgKey
+// @Description  Fetch gpgkey from URL
+// @Tags         gpgKey
+// @Accept       json
+// @Produce      json
+// @Success      200 {object} api.FetchGPGKeyResponse
+// @Router       /repository_parameters/external_gpg_key [get]
+func (rh *RepositoryParameterHandler) fetchGpgKey(c echo.Context) error {
+	url := c.QueryParam("url")
+
+	if url == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "url cannot be blank")
+	}
+
+	gpgKey, err := rh.ExternalResourceDao.FetchGpgKey(url)
+
+	if err != nil {
+		httpError := echo.NewHTTPError(http.StatusNotAcceptable, "Received response was not a valid gpg-key")
```

Comment:
Is this response useful for the user? 
![Screen Shot 2022-09-28 at 3 16 31 PM](https://user-images.githubusercontent.com/38083295/192893920-fd079aa6-91a7-4065-a284-37d23d028b59.png)

Reply:
GPG Key :)